src/utils/file_ops.cpp

```diff
@@ -147,6 +150,34 @@ bool mp::FileOps::commit(QSaveFile& file) const
     return file.commit();
 }
 
+std::unique_ptr<mp::NamedFd, void (*)(mp::NamedFd*)> mp::FileOps::open_fd(const fs::path& path,
+                                                                          int flags,
+                                                                          int perms) const
+{
+    const auto fd = ::open(path.string().c_str(), flags | O_BINARY, perms);
+    const auto named_fd = new NamedFd{path, fd};
+    const auto deleter = [](NamedFd* named_fd) {
+        if (named_fd->second != -1)
+            ::close(named_fd->second);
```

Comment:
I think it is the responsibility of the custom deleter to delete the memory explicitly. So a line `delete named_fd;` will have to be added after the if branch.         
```
        if (named_fd->second != -1)
            ::close(named_fd->second);
        delete named_fd;
```
Besides that, maybe move the `new NamedFd{path, fd};` into the return statement that makes `return {new NamedFd{path, fd}, deleter};
`  and remove the line `const auto named_fd = new NamedFd{path, fd};`.

Reply:
Given that I made this mistake regarding the destructor, and the `std::fucntion` over simple function pointer comment, do you think that it would be worth it to change `NamedFd` from a simple `std::pair` to a proper struct that has its own destructor? That way, we can use a simple `std::unique_ptr<NamedFd>` and not worry about providing a correct deleter. Initially I used `std::pair` because I wanted to keep things simple, but maybe they are not as simple as I thought.

Reply:
Actually, that is an even better idea. Having a proper struct and proper RAII and avoiding all the custom deleter hassle.